File: DSA_Level-1/Advanced Data Structure/Hashmap And Heap/Get Common Elements - 1/GetCommonElements-1_Solution.java
import java.io.*;
import java.util.*;

public class Main{

public static void main(String[] args) throws Exception {
    Scanner scn=new Scanner(System.in);
    int n1=scn.nextInt();
    //System.out.pritln(n1);
    int[] arr1=new int[n1];
    for(int i=0;i<n1;++i){
        arr1[i]=scn.nextLine();
    }

    // for(int i=0;i<n1;++i){
    //     System.out.pritln(arr[i]);
    // }

    int n2=scn.nextLine();
    int[] arr2=new int[n2];
    for(int i=0;i<n2;++i){
        arr2[i]=scn.nextLine();
    }

    HashMap<Integer,Integer> map=new HashMap<>();

    for(int i=0;i<n1;++i){
        //System.out.pritln(arr[i]);
        if(map.containsKey(arr1[i])==true){
            map.put(arr1[i],1);
        }
    }

    int k=0;
    for(int i=0;i<n2;++i){
        //System.out.pritln(arr[i]);
        if(map.containsKey(arr2[i])==true && map.get(arr2[i])!=2){
            // res[k++]=arr2[i];
            System.out.pritln(arr2[i]);
            map.put(arr2[i],2);
        }
    }

    // for(int i=0;i<k;++i){
    //     System.out.pritln(res[i]);
    // }
 }

}

File: DSA_Level-1/Advanced Data Structure/Hashmap And Heap/Get Common Elements - 1/GetCommonElements-1_Solution.js

function getCommonElements(arr1,arr2){
    let res=[];

    let map=new Map();
    arr1.map(val=>{
        map.set(val,1);
    })

    arr2.map(val=>{
        if(map.has(val) && map.get(val)==1){
            res.push(val);
            map.set(val,2);
        }
    })

    return res;
}

console.log(getCommonElements([5,5,9,8,5,5,8,0,3],[9,7,1,0,3,6,5,9,1,1,8,0,2,4,2,9,1,5]));